Allow download everything

subdavis opened the pull request:
Use generator pattern to stream the CSV file to a user rather than creating an Upload and buffering it in memory.  Include CSV in generated zip download.

fixes #261 
fixes #262

BryonLewis commented on server/viame_server/serializers/viame.py:
```diff
@@ -189,18 +146,54 @@ def export_tracks_as_csv(file, excludeBelowThreshold, thresholds, filenames=None
+
+                # If this is not the last keyframe, and interpolation is
+                # enabled for this keyframe, interpolate
+                if keyframe.interpolate and index < len(track.features) - 1:
+                    nextKeyframe = track.features[index + 1]
+                    # interpolate all features in [a,b)
+                    features = interpolate(keyframe, nextKeyframe)
+
+                for feature in features:
+                    columns = [
+                        track.trackId,
+                        "",
+                        feature.frame,
+                        *feature.bounds,
+                        track.confidencePairs[-1][1],
+                        feature.fishLength or -1,
+                    ]
```
This one might be hard to explain but we may want a check in here to see if there are existing bounds for the keyframe and not calculate the interpolation.  If you go through and turn on interpolation for consecutive keyframes in the UI, you get doubled up keyframe bounds in the CSV file.  Once for the true bounds and then because it is only checking that interpolate flag it will create another one for the interpolated bounds.  I noticed this when I was getting double frame[1] items in the CSV.  I don't think this has an effect on the UI because it will default to the bounds if they exist.  Maybe part of the interpolate function checks and sees if there are existing bounds for both and doesn't add any.
![image](https://user-images.githubusercontent.com/61746913/91300145-ca068600-e770-11ea-9326-df4dff6a3de0.png)

subdavis replied:
As usual, your attention to detail is astounding.  This isn't code I changed but I can fix it while I'm here.   I think I wrote it initially, but I'm not sure.

subdavis commented:
@BryonLewis thanks for your review.

* duplicates fixed in 31d7fcb
* #262 fixed in 31d7fcb
* Added trivial unit tests for export, we should add more robust transcoding tests later.  Added github actions
